datadog_checks_dev/datadog_checks/dev/tooling/configuration/spec.py

```diff
@@ -0,0 +1,588 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from .utils import default_option_example, normalize_source_name
+
+
+def spec_validator(spec, loader):
+    if not isinstance(spec, dict):
+        loader.errors.append('{}: Configuration specifications must be a mapping object'.format(loader.source))
+        return
+
+    if 'name' not in spec:
+        loader.errors.append(
+            '{}: Configuration specifications must contain a top-level `name` attribute'.format(loader.source)
+        )
+        return
+
+    name = spec['name']
+    if not isinstance(name, str):
+        loader.errors.append('{}: The top-level `name` attribute must be a string'.format(loader.source))
+        return
+
+    if 'version' not in spec:
```

Comment:
1. Should the `version` be optional (or even added later) until we introduce a new version?

2. Isn't `semver` somewhat overkill for schema version. Usually the schema/manifest versions used are single or 2 digit. e.g. docker-compose, Dockerfile, circleci yaml, etc. 

The version `patch` part seems not useful for configuration format versioning since fixing a bug (`patch`) doesn't affect the version format itself. Only `minor` for improvements, and `major` for breaking changes have meaning in this case.

Reply:
The version mirrors the check version and should be bumped during releases. I already added a card

Reply:
If it mirrors the version of the check. What the purpose of the version then?

Reply:
It's mostly useful for accurate validation/generation of config given a version

Reply:
> It's mostly useful for accurate validation/generation of config given a version

I'm not sure to follow, the configuration specs are in VCS with the integrations code itself, so it's always in sync. I'm not sure how the version help or is useful. Maybe I'm missing something.

Reply:
1. When we begin storing the spec Somewhere ™️ , we don't yet know if the upload has the context to determine the version based on yet another file in the repo or if it will just receive the validated spec output
2. It would be nice for users to immediately see what version of an integration they are configuring by merely looking at the top of the example conf

Reply:
Sure, I see. It's fine, if it's automatically updated/bumped.